test/trento/activity_logging/activity_catalog_test.exs

```diff
@@ -1,21 +1,76 @@
 defmodule Trento.ActivityLog.ActivityCatalogTest do
   @moduledoc false
 
+  import Trento.Factory
+
   use TrentoWeb.ConnCase, async: false
 
   alias Trento.ActivityLog.ActivityCatalog
 
   require Trento.ActivityLog.ActivityCatalog, as: ActivityCatalog
 
+  describe "activity detection" do
+    test "should not detect activity from invalid input" do
+      Enum.each([nil, %{}, "", 42], fn input ->
+        assert nil == ActivityCatalog.detect_activity(input)
+      end)
+    end
+
+    test "should not be able to detect an activity from a connection without controller/action info",
+         %{conn: conn} do
+      assert nil == ActivityCatalog.detect_activity(conn)
+    end
+
+    test "should detect activity from a connection with controller/action info", %{conn: conn} do
+      assert {Foo.Bar.AcmeController, :foo_action} ==
+               conn
+               |> Plug.Conn.put_private(:phoenix_controller, Foo.Bar.AcmeController)
+               |> Plug.Conn.put_private(:phoenix_action, :foo_action)
+               |> ActivityCatalog.detect_activity()
+    end
+
+    test "should detect activity from domain events" do
+      %event_module{} = event = build(:host_registered_event)
+      metadata = %{}
+      assert event_module == ActivityCatalog.detect_activity(%{event: event, metadata: metadata})
+    end
+  end
+
   test "should ignore unknown activities", %{
     conn: conn
   } do
-    Enum.each([:foo_bar, %{bar: "baz"}, "not-interesting"], fn activity ->
+    Enum.each([%{bar: "baz"}, "not-interesting", %{}, 42], fn activity ->
       refute ActivityCatalog.interested?(activity, conn)
       assert nil == ActivityCatalog.get_activity_type(activity)
     end)
   end
 
+  describe "activity type detectiion" do
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**typo:** 
‏
```suggestion
  describe "activity type detection" do
```